### ISSUE TITLE
When "core" and "bgc" are chosen in mapApp() use synthetic index

### ISSUE THREAD

**Issue by j-harbin** [ArgoCanada/argoFloats#548]
Currently in `mapApp` when "core" and "bgc" are selected, we are getting the "core" and the "bgc" index and merging them together. Once I read up some more on the synthetic files at http://www.argodatamgt.org/Data-Mgt-Team/News/BGC-Argo-synthetic-profiles-distributed-on-GDAC  my guess is I will likely change this behaviour to get the synthetic index instead.

**Comment by j-harbin** [comment]
I'm going to keep this as is. Based on the documentation in the link above, "For each cycle, the existing Argo NetCDF V3.1 profile file contains one or more core-Argo profile (pressure, temperature, salinity) and one or more additional BGC profile (oxygen, chlorophyll, etc…)." 

My take from this is that if a core float solely contained pressure, temperature, and salinity (and no BGC measurements) they would not show up in synthetic data. I tested this theory by looking at an early float in 2000 (before BGC floats were introduced). The output is below which tells us this theory is true.

``` r
library(argoFloats)
ai <- getIndex()
index1 <- subset(ai, ID="13857") # Early float before BGC floats were introduced
#> Kept 138 cycles (0.00528%)
argos <- readProfiles(getProfiles(index1))
#> Warning in readProfiles(getProfiles(index1)): Of 138 profiles read, 1 has >10% of temperature values with QC flag of 4, signalling bad data.
#>     The indices of the bad profiles are as follows.
#>     138
sort(names(argos[[1]][['data']])) # Shows only core measurements
#> [1] "latitude"                 "longitude"               
#> [3] "pressure"                 "pressureAdjusted"        
#> [5] "pressureAdjustedError"    "temperature"             
#> [7] "temperatureAdjusted"      "temperatureAdjustedError"
#> [9] "time"
sai <- getIndex(filename="synthetic")
index2 <- subset(sai, ID="13857")
#> Kept 0 cycles (0%)
```

<sup>Created on 2022-04-04 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

In summary, if both "core" and "bgc" floats are selected, and I simply use `getIndex(filename="synthetic")` I would be missing floats that don't contain both core and BGC measurements.
